src/sqlparser/tests/testdata/array.yaml

```diff
@@ -74,3 +74,6 @@
     sql parser error: expected an expression, found: [
     LINE 1: SELECT [1,2]
                    ^
+- input: CREATE TABLE t (params STRUCT<a STRUCT<b int>>[])
```

Comment:
This test case seems to work in current version. Did I miss something?
```
dev=> CREATE TABLE t (params STRUCT<a STRUCT<b int>>[])
dev-> ;
CREATE_TABLE
dev=> describe t;
┌───────────────────┬─────────────────────┬───────────┬─────────────┐
│       Name        │        Type         │ Is Hidden │ Description │
├───────────────────┼─────────────────────┼───────────┼─────────────┤
│ params            │                     │ false     │ ∅           │
│ params.a          │ struct<b integer>[] │ false     │ ∅           │
│ _row_id           │ serial              │ true      │ ∅           │
│ primary key       │ _row_id             │ ∅         │ ∅           │
│ distribution key  │ _row_id             │ ∅         │ ∅           │
│ table description │ t                   │ ∅         │ ∅           │
└───────────────────┴─────────────────────┴───────────┴─────────────┘
(6 rows)
```

Reply:
Oh, it's wrong

Reply:
How did you find the bug? Is it related with other bugs?

Reply:
See the linked issue. It was causing a schema mismatch panic when `sink into table`.

Reply:
Oh, I didn't notice there's linked issue since there's no `fix #xx` in body 😄